readme.md

```diff
@@ -615,6 +615,116 @@ The order where these resources are used are:
+  }  
+  
+}
+```
+the `application.properties`
+```
+dekorate.component.name=hello-annotationless-world  
+dekorate.component.envs[0].name=key_from_properties  
+dekorate.component.envs[0].value=value_from_properties  
+dekorate.component.deploymentMode=build
+``` 
+the result of this configuration:
+```
+---  
+apiVersion: "v1"  
+kind: "List"  
+items:  
+- apiVersion: "halkyon.io/v1beta1"  
+  kind: "Component"  
+  metadata:  
+    labels:  
+      app: "hello-annotationless-world"  
+  name: "hello-annotationless-world"  
+  spec:  
+    deploymentMode: "build"  
+  runtime: "spring-boot"  
+  version: "2.1.6.RELEASE"  
+  exposeService: false  
+  envs:  
+    - name: "key_from_properties"  
+    value: "value_from_properties"  
+  buildConfig:  
+      type: "s2i"  
+      url: "https://github.com/dekorateio/dekorate.git"  
+      ref: "master"  
+      contextPath: "examples/"  
+      moduleDirName: "halkyon-example-annotationless-properties"
+```
+Note that some items does not end up in the same hierarchical place where they were configured, in the case of `deploymentMode` an additional level `spec` has been introduced.
+You can find [here](https://github.com/dekorateio/dekorate/blob/master/examples/halkyon-example-annotationless-properties/src/main/resources/application.properties) the code of this example.
+###### how to override annotationbased-config
+Given a spring boot application class:
+```
+package io.dekorate.examples.component;  
+  
+import io.dekorate.halkyon.annotation.HalkyonComponent;  
+import io.dekorate.kubernetes.annotation.Env;  
+import org.springframework.boot.SpringApplication;  
+import org.springframework.boot.autoconfigure.SpringBootApplication;  
+  
+@HalkyonComponent(name = "halkyon", exposeService = true, envs = @Env(name = "key1", value = "val1"))  
+@SpringBootApplication  
+public class Application {  
+  
+  public static void main(String[] args) {  
+    SpringApplication.run(Application.class, args);  
+  }  
+}
+```
+If we provide an `application.yml` file as follows:
+```
+dekorate:  
+  component:  
+    name: "hello-world"  
+    buildType: "docker"  
+    deploymentMode : build
+```
+the halkyon component crd will contain the configuration provided by file:
+```
+---  
+apiVersion: "v1"  
+kind: "List"  
+items:  
+- apiVersion: "halkyon.io/v1beta1"  
+  kind: "Component"  
+  metadata:  
+    labels:  
+      app: "hello-world"  
+  version: "0.0.1-SNAPSHOT"  
+  name: "hello-world"  
+  spec:  
+    deploymentMode: "build"  
+  runtime: "spring-boot"  
+  version: "2.1.6.RELEASE"  
+  exposeService: false  
+  buildConfig:  
+    type: "docker"  
+    url: "https://github.com/dekorateio/dekorate.git"  
+    ref: "master"  
+    contextPath: "annotations/halkyon-annotations/target/it/"  
+    moduleDirName: "feat-229-override-annotationbased-config"
+
+```
+As in the previous example an additional level `spec` has been introduced for `deploymentMode`. 
```

Comment:
We should insist on the fact that the annotation-less configuration completely overrides the annotation configuration as is evidenced by the fact that `exposeService` becomes `false` despite being configured in the annotation… Maybe that needs to be explained above when we say that the annotation-less mode takes precedence, which is note as clear as it should be, imo. 
I think this precedence point will cause issues with our users because I think the natural behavior most people will expect is a merge behavior.

Reply:
I guess this answers my question [here](https://github.com/dekorateio/dekorate/issues/313#issuecomment-523367339)